datasource/common/src/main/java/org/opennms/oce/datasource/common/SituationBean.java

```diff
@@ -97,15 +99,15 @@ public void addResourceKey(ResourceKey resourceKey) {
 
     @Override
     public Set<Alarm> getAlarms() {
-        return alarms;
+        return alarms.values().stream().collect(Collectors.toSet());
     }
 
     public void setAlarms(Set<Alarm> alarms) {
-        this.alarms = alarms;
+        this.alarms = alarms.stream().collect(Collectors.toMap(Alarm::getId, a -> a));
```

Comment:
To continue using the sorted map implementation:

```suggestion
        this.alarms.clear();
        this.alarms.putAll(alarms.stream().collect(Collectors.toMap(Alarm::getId, a -> a)));
```